### ISSUE TITLE
be able to add state into the link like react router

### ISSUE THREAD

**Issue by numbing** [molefrog/wouter#316]
in react router we can add state into the Link that work like this:
`<Link to={{pathname: '/nextpath', state: { prevPath: location.pathname }}}>Example Link</Link>`

it would be great if had a feature that work like this in wouter

**Comment by redwanul10** [comment]
Hi @numbing ! I have made a PR [#325](https://github.com/molefrog/wouter/pull/325)  let's wait for @molefrog to review :smiley:

**Comment by molefrog** [comment]
Hey @redwanul10 thank you for your PR. We made a complete revamp of the library and state support is now available in v3.0.0-rc.1 (see my comment https://github.com/molefrog/wouter/issues/102#issuecomment-1822253686)

We also made a standalone hook for getting the state:

```js
import { useHistoryState } from "wouter/use-browser-location"

// this will only re-render when the actual state changes, not the URL
const state = useHistoryState()
```

It also uses `useSyncExternalStore` under the hood, so it solves the SSR issue you had.
